src/System.Runtime.Extensions/src/System/IO/Path.Unix.cs

```diff
@@ -209,5 +210,11 @@ private static unsafe byte[] CreateCryptoRandomByteArray(int byteLength)
             }
             return arr;
         }
+
+        private static bool HasIllegalCharacters(string path, bool checkAdditional)
+        {
+            Contract.Requires(path != null);
+            return path.IndexOfAny(checkAdditional ? InvalidPathCharsWithAdditionalChecks : InvalidPathChars) >= 0;
```

Comment:
Once this moves here, we can get rid of the InvalidPathCharsWithAdditionalChecks field from the Unix implementation, and just simplify this down to ```return path.IndexOfany(InvalidPathChars) >= 0;```.